posthog/queries/test/test_trends.py

```diff
@@ -166,6 +168,210 @@ def test_trends_compare(self):
             self.assertEqual(no_compare_response[0]["labels"][5], "Thu. 2 January")
             self.assertEqual(no_compare_response[0]["data"][5], 1.0)
 
+        def _test_interval_filtering(self, dates: List[str], result, **filter_params):
```

Comment:
stricly implementing these tests is already showing discrepancies between postgres and clickhouse query results that weren't considered before.